newrelic-alerts-configurator/src/main/java/com/ocadotechnology/newrelic/alertsconfigurator/NrqlConditionConfigurator.java

```diff
@@ -0,0 +1,123 @@
+package com.ocadotechnology.newrelic.alertsconfigurator;
+
+import com.ocadotechnology.newrelic.alertsconfigurator.configuration.PolicyConfiguration;
+import com.ocadotechnology.newrelic.alertsconfigurator.configuration.condition.nrql.NrqlCondition;
+import com.ocadotechnology.newrelic.alertsconfigurator.configuration.condition.nrql.NrqlUtils;
+import com.ocadotechnology.newrelic.alertsconfigurator.configuration.condition.terms.TermsUtils;
+import com.ocadotechnology.newrelic.alertsconfigurator.exception.NewRelicSyncException;
+import com.ocadotechnology.newrelic.apiclient.NewRelicApi;
+import com.ocadotechnology.newrelic.apiclient.model.conditions.nrql.AlertsNrqlCondition;
+import com.ocadotechnology.newrelic.apiclient.model.policies.AlertsPolicy;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static java.lang.String.format;
+
+@Slf4j
+public class NrqlConditionConfigurator {
```

Comment:
This class is copy of `ConditionConfigurator` with condition replaced by nrql condition. Whole logic of synchronizing conditions is duplicated here. Common code should be extracted and reused. The same applies to test of these two classes.

Reply:
`ExternalServiceConfigurator` is also very similar. I'll commonize all Configurators